ten/one_david.py

```diff
@@ -0,0 +1,76 @@
+class BinaryHeap:
+    
+    def __init__(self, min=True):
+        self.arr = []
+        self.size = 0
```

Comment:
I would have written a `size` method instead which just returns `len(self.arr)` to avoid having to keep track of this separately.

Reply:
needed the separate size tracker because i don't shrink the array after i pop an entry